csharp/Squidex.ClientLibrary/Squidex.ClientLibrary/ContentsClient.cs

```diff
@@ -106,23 +106,21 @@ public Task<TEntity> CreateAsync(TData data, bool publish = false, CancellationT
             return RequestJsonAsync<TEntity>(HttpMethod.Post, BuildSchemaUrl($"?publish={publish}", false), data.ToContent(), ct: ct);
         }
 
+        public Task<TEntity> CreateDraftAsync(Guid id, CancellationToken ct = default)
+        {
+            Guard.NotEmpty(id, nameof(id));
+
+            return RequestJsonAsync<TEntity>(HttpMethod.Post, BuildSchemaUrl($"{id}/draft", false), ct: ct);
+        }
+
         public Task<TEntity> UpdateAsync(Guid id, TData data, bool asDraft = false, CancellationToken ct = default)
         {
             Guard.NotEmpty(id, nameof(id));
             Guard.NotNull(data, nameof(data));
 
             if (asDraft)
             {
-                QueryContext ctx = QueryContext.Default.Unpublished(true);
-                TEntity content = this.GetAsync(id, ctx).Result;
-                if (content.Status == Status.Published)
-                {
-                    content = RequestJsonAsync<TEntity>(HttpMethod.Post, BuildSchemaUrl($"{id}/draft", false), data.ToContent(), ct: ct).Result;
-                    if (content.Status != Status.Draft)
-                    {
-                        throw new SquidexException($"Failed to create draft for content id: {id}");
-                    }
-                }
+                TEntity content = CreateDraftAsync(id, ct).Result;
```

Comment:
you are not awaiting it and what are you doing with the result object?

Reply:
I can't "await" this call, as the UpdateAsync method is not marked as async, therefore the code is effectively waiting by assigning the Result to a variable... more accurately, I should have done:
CreateDraftAsync(id, ct).GetAwaiter().GetResult();

Reply:
Just mark it as async.

Reply:
Done